### PR TITLE
Set up dependabot for automated Dockerfile updates.

### PR DESCRIPTION
**Pull request by fniephaus**
## What does this PR do?

This PR adds a `dependabot.yml` config that enables automatic Dockerfile version updates, and adds Dockerfiles for all allowed Docker images.

## Checklist before merging

n/a

### CODE REVIEW

**Comment by dnestoro** [comment]
LGTM! Next steps would be changing DockerTask and GrypeTask so we can get rid of the list. I can do that. What do you think @fniephaus should we do that in a follow-up PR or I should contribute to this PR so we can merge everything together?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+version: 2
+
+updates:
+  - package-ecosystem: "docker"
+    directory: "/tests/tck-build-logic/src/main/resources/allowed-docker-images"
+    reviewers:
+      - "matneu"
+    schedule:
+      interval: "weekly"
```
Comment (matneu) [inline .github/dependabot.yml:12]: Any reason we pick weekly? I think this can be a daily pass, shouldn't take long?

Comment (fniephaus) [reply .github/dependabot.yml]: Sure, can be daily.

**Comment by fniephaus** [comment]
> What do you think @fniephaus should we do that in a follow-up PR or I should contribute to this PR so we can merge everything together?

I don't think we need to do everything in this PR. Let's get rid of the `AllowedDockerImages.txt` in a follow-up PR.

**Review by dnestoro — APPROVED** [review]

**Review by matneu — APPROVED** [review]

**Pull request merged**